src/stake-pool/utils.ts

```diff
@@ -671,6 +744,15 @@ export function calcStakeDeposit(
   );
 }
 
+export function ceilDiv(
+  numerator: Numberu64,
+  denominator: Numberu64,
+): Numberu64 {
+  return Numberu64.cloneFromBN(
+    numerator.add(denominator).sub(new Numberu64(1)).div(denominator),
+  );
+}
```

Comment:
so this isnt being used anymore? Do you wanna try doing the refactor again and seeing if there's still issues?